sdk/python/kfmd/metadata.py

```diff
@@ -82,21 +83,26 @@ def log(self, artifact):
     This method expects `artifact` to have
       - ARTIFACT_TYPE_NAME stirng field the form of
         /artifact_types/<namespace>/<name>.
-      - serialization() method to return a swagger_client.ApiArtifact.
+      - serialization() method to return a openapi_client.MlMetadataArtifact.
 
-    This method will set artifact._id and artifact._create_time.
+    This method will set artifact.id.
     """
 
     # TODO(zhenghui): log this artifact as the input or output of an execution.
     serialization = artifact.serialization()
-    if serialization.workspace.name == None:
-      serialization.workspace.name = self.workspace.name
+    if serialization.custom_properties == None:
+          serialization.custom_properties = {}
+    serialization.custom_properties[
+      "__workspace__"] = openapi_client.MlMetadataValue(
```

Comment:
should we use `__kf_workspace` ? And check that clients never use this property accidentally?

Reply:
Done.